presto-hive/src/main/java/io/prestosql/plugin/hive/metastore/thrift/DefaultThriftMetastoreClientFactory.java

```diff
@@ -120,16 +124,27 @@ private TTransport createTransport(HostAndPort address, Optional<String> delegat
         try {
             // load KeyStore if configured and get KeyManagers
             KeyManager[] keyManagers = null;
+            KeyStore keyStore = null;
+            char[] keyManagerPassword = new char[0];
             if (keyStorePath.isPresent()) {
-                KeyStore keyStore = PemReader.loadKeyStore(keyStorePath.get(), keyStorePath.get(), keyStorePassword);
+                try {
+                    keyStore = PemReader.loadKeyStore(keyStorePath.get(), keyStorePath.get(), keyStorePassword);
+                }
+                catch (IOException | GeneralSecurityException ignored) {
```

Comment:
Is it meaningful to ignore the `GeneralSecurityException`? For example, trying the same password would fail again if the `keyStorePassword` is wrong. 

Is it possible to catch only the exception which gives us another chance to read the key store directly? Or it should at least show some warning or error message if the exception happens.

Reply:
Couldn't find a particular exception which should tell us to read the key store directly. Added warning log.